client/interface_wireguard_test.go

```diff
@@ -53,19 +53,21 @@ func TestAddFindDeleteInterfaceWireguard(t *testing.T) {
 	if !reflect.DeepEqual(created, found) {
 		t.Error("expected created and found resources to be equal, but they don't")
 	}
-	err = c.Delete(found.(Resource))
-	if err != nil {
-		t.Errorf("expected no error, got %v", err)
-	}
+	defer func() {
+		err = c.Delete(found.(Resource))
+		if err != nil {
+			t.Errorf("expected no error, got %v", err)
+		}
 
-	_, err = c.Find(findInterface)
-	if err == nil {
-		t.Errorf("expected error, got nothing")
-		return
-	}
+		_, err = c.Find(findInterface)
+		if err == nil {
+			t.Errorf("expected error, got nothing")
+			return
+		}
 
-	target := &NotFound{}
-	if !errors.As(err, &target) {
-		t.Errorf("expected error to be of type %T, got %T", &NotFound{}, err)
-	}
+		target := &NotFound{}
+		if !errors.As(err, &target) {
+			t.Errorf("expected error to be of type %T, got %T", &NotFound{}, err)
+		}
+	}()
```

Comment:
I'm pretty confident the placement of the `defer` matters. For example, an error from [line 43](https://github.com/ddelnano/terraform-provider-mikrotik/pull/153/commits/6e3f449c7186d0222e7cb4d20fa9a0a6980a3476#diff-a2906da3a2bfe7592ffee554d7416a71d9efe76a43106e90226341cdfdb0f142R43) will cause the `defer` to get skipped completely.

We should place the `defer` right after the creation happens to ensure it is registered after the creation completes.

Reply:
[Here's](https://go.dev/play/p/B6qi05kn0KF) an example that showcases this.

<img width="907" alt="Screen Shot 2023-05-13 at 12 20 12 AM" src="https://github.com/ddelnano/terraform-provider-mikrotik/assets/5855593/59749e07-203e-4e39-821a-c5494e224a33">